### PR TITLE
fix(workbench-header): allow ReactNode as title

### PR DESCRIPTION
**Pull request by denkristoffer**
# Purpose of PR

We're currently using elements in the WorkbenchHeader `title` in some places, the current types don't allow for that.

Also adds some other fixes needed for the content model editor:
- `icon` prop takes an Element, which is needed for passing things like `variant` or `size` on product icons.

### CODE REVIEW

**Comment by denkristoffer** [comment]
I think this might be too simple a solution, since the current Header wraps `title` in a `Heading` component 🤔

File: packages/components/workbench/src/WorkbenchHeader/WorkbenchHeader.tsx [hunk 1]
```diff
@@ -10,7 +11,7 @@ import { getWorkbenchHeaderStyles } from './WorkbenchHeader.styles';
 
 export interface WorkbenchHeaderProps extends CommonProps {
   /** This is the title that will be shown inside the Header component */
-  title: string;
+  title: ReactNode;
```
Comment (gui-santos) [inline packages/components/workbench/src/WorkbenchHeader/WorkbenchHeader.tsx:14]: Maybe you could include the change in line 61 in this PR

Basically, if title is a string we render the way we render it now, but if is a ReactNode we just render the ReactNode

wdyt?

**Review by gui-santos — DISMISSED** [review]

File: packages/components/workbench/src/WorkbenchHeader/WorkbenchHeader.tsx (hunk 1 above)
Comment (gui-santos) [reply packages/components/workbench/src/WorkbenchHeader/WorkbenchHeader.tsx:14]: @denkristoffer any chance of following up with this PR changes?

**Review by bgutsol — COMMENTED** [review]
Great, looks good!
Can you also add a small description and examples to the README file? With the example of title and icon passed as a ReactElement.

**Comment by denkristoffer** [comment]
@bgutsol Sure, but is the Workbench from Forma deprecated or not? I see that there's no existing examples or much documentation for it.

**Comment by bgutsol** [comment]
> @bgutsol Sure, but is the Workbench from Forma deprecated or not? I see that there's no existing examples or much documentation for it.

@denkristoffer we have [docs page](https://f36.contentful.com/components/workbench), here is the [readme](https://github.com/contentful/forma-36/blob/607301d57a2e83190d2aa298120ddb8493e8c429/packages/components/workbench/README.mdx). 
But you are right there are no usage examples. @gui-santos do you know why we decided not to add any examples.

@denkristoffer Not to block this PR, maybe let's just add examples in the storybook, at least for future us. WDYT?

**Review by bgutsol — APPROVED** [review]
Thanks @denkristoffer

**Review by thebesson — APPROVED** [review]
@denkristoffer Workbench is being deprecated, this is why it's not included in the main package, just FYI 👍

**Comment by gui-santos** [comment]
> 
> But you are right there are no usage examples. @gui-santos do you know why we decided not to add any examples.
> 

Because we were supposed to remove Workbench from everywhere, but I don't think we will do that any time soon so let's add examples to the docs!

**Pull request merged**